src/sources/opentelemetry/integration_tests.rs

```diff
@@ -0,0 +1,133 @@
+use std::time::Duration;
+
+use futures::Stream;
+use futures_util::StreamExt;
+use serde_json::json;
+
+use crate::{
+    config::{log_schema, SourceConfig, SourceContext},
+    event::{into_event_stream, Event, EventStatus},
+    test_util::{
+        collect_n,
+        components::{assert_source_compliance, HTTP_PUSH_SOURCE_TAGS},
+        retry_until, wait_for_tcp,
+    },
+    SourceSender,
+};
+
+use super::{GrpcConfig, HttpConfig, OpentelemetryConfig, LOGS};
+
+fn otel_health_url() -> String {
+    std::env::var("OTEL_HEALTH_URL").unwrap_or_else(|_| "http://0.0.0.0:13133".to_owned())
+}
+
+fn otel_otlp_url() -> String {
+    std::env::var("OTEL_OTLPHTTP_URL").unwrap_or_else(|_| "http://0.0.0.0:9876".to_owned())
+}
+
+fn source_grpc_address() -> String {
+    std::env::var("SOURCE_GRPC_ADDRESS").unwrap_or_else(|_| "0.0.0.0:4317".to_owned())
+}
+
+fn source_http_address() -> String {
+    std::env::var("SOURCE_HTTP_ADDRESS").unwrap_or_else(|_| "0.0.0.0:4318".to_owned())
+}
+
+#[tokio::test]
```

Comment:
Are there any error paths that are worth testing ?

Reply:
Unhappy path is _difficult_ in the integration tests here, since everything coming out of the opentelemetry-collector should be valid and properly formatted. I think the biggest integration concern would be ensuring the protobuf definition is the same between our implementation and theirs (and/or an SDK when one exists).

Reply:
We could add some unit tests to ensure the requests/responses are properly shaped.

Reply:
I'll include these tests in the upcoming work for otel metrics which will use the same serve and requests/responses